canonical-web-and-design/maas-ui
# Fix bootstrapping error

## Done

- Move authentication checking into single-spa lifecycle.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)

### QA steps

- Open a private Firefox window.
- Load a legacy url e.g. /MAAS/l/domains.
- You should get redirected to the login form. There shouldn't be a console error: `Error: application 'legacy' died in status BOOTSTRAPPING: {"isTrusted":true}`.
- Log in and navigate to the legacy app, which should load correctly.

## Fixes

Fixes: https://github.com/canonical-web-and-design/maas-ui/issues/1459.
